highlight/highlight
# Allow users to sort traces (take 2)

## Summary

This PR reverts the revert of #8497 and introduces some fixes + improvements.

* We started using the `HighlightType` column rather than `TraceAttributes['highlight.type']` for default trace filters, which should provide a performance improvement.
* We only hit the sampling table for the inner query when selecting traces.
* Adds the sort param to the queries fetching additional results.

## How did you test this change?

* Load the traces page and ensure results + stats in the top look correct
  * Inspect SQL statements to ensure the `traces` table is being hit
* Click on a column header to sort results
  * Ensure results are sorted correctly
  * Inspect SQL and ensure the `traces_sampling` table is hit for the inner query and `traces` is hit for the outer query
* Ensure all other querying behavior works as expected, including logs

## Are there any deployment considerations?

Will monitor as this goes out again to ensure it doesn't cause any more issues.

## Does this work require review from our design team?

Will have @julian-highlight click test in prod. We may do some follow up on the UX to trigger the dropdown menus on table headers.
